full-moon/src/ast/parser_util.rs

```diff
@@ -179,6 +179,12 @@ macro_rules! keep_going {
 #[doc(hidden)]
 #[macro_export]
 macro_rules! define_roblox_parser {
+    ($parser:ident, $node:ty, $mock_ty:ty, |_, $state:ident| $body:expr) => {
+        define_roblox_parser! {$parser, $node, $mock_ty, |_, mut $state: ParserState<'a, 'b>| $body}
+    };
+    ($parser:ident, $node:ty, $mock_ty:ty, |$self:ident, $state:ident| $body:expr) => {
+        define_roblox_parser! {$parser, $node, $mock_ty, |$self:&$parser, mut $state: ParserState<'a, 'b>| $body}
```

Comment:
I'm confused, how does this work? This doesn't appear to be a valid definition of `define_roblox_parser`, all the current definitions use `()`.

Reply:
I'm not entirely sure what you're referring to here, but I just copied and modified the layout of the define_parser macro to fit here

Reply:
I'm referring to `define_roblox_parser! {}` using braces instead of parentheses, when I don't see that being a valid macro format.

Reply:
It is a valid format. The reason its using `{}` instead of `()` is that the macro expands to items.
The alternative could be to leave it as `()` and add a semicolon to the end, but this way works fine and is valid.
I quickly googled and its mentioned here; this document is old but it still seems to be valid: http://web.mit.edu/rust-lang_v1.25/arch/amd64_ubuntu1404/share/doc/rust/html/book/first-edition/macros.html#syntactic-requirements

Reply:
Just looking at the change, it seems rustfmt formats it into something quite ugly...
Should we add a skip attribute, revert to the original, or just leave it?